### PR TITLE
[TACHYON-1412] Refactor PinListSync to implement HeartbeatExecutor

### PR DESCRIPTION
**Pull request by shuhuai007**
Please reivew:
https://tachyon.atlassian.net/browse/TACHYON-1412

### CODE REVIEW

**Comment by yupeng9** [comment]
LGTM. Thanks for the improvement!

File: common/src/main/java/tachyon/heartbeat/HeartbeatContext.java [hunk 1]
```diff
@@ -46,6 +47,7 @@
     sTimerClasses.put(WORKER_BLOCK_SYNC, SLEEPING_TIMER_CLASS);
     sTimerClasses.put(WORKER_LINEAGE_SYNC, SLEEPING_TIMER_CLASS);
     sTimerClasses.put(WORKER_CLIENT, SLEEPING_TIMER_CLASS);
+    sTimerClasses.put(PIN_LIST_SYNC, SLEEPING_TIMER_CLASS);
```
Comment (aaudiber) [inline common/src/main/java/tachyon/heartbeat/HeartbeatContext.java]: Rename to `WORKER_PIN_LIST_SYNC` to be consistent. Also please alphabetize, we're so close!

File: common/src/test/java/tachyon/heartbeat/HeartbeatContextTest.java [hunk 2]
```diff
@@ -36,6 +36,7 @@ public void timerClassesCheckTest() throws Exception {
     checkNotNull(HeartbeatContext.MASTER_TTL_CHECK);
     checkNotNull(HeartbeatContext.WORKER_CLIENT);
     checkNotNull(HeartbeatContext.WORKER_LINEAGE_SYNC);
+    checkNotNull(HeartbeatContext.PIN_LIST_SYNC);
```
Comment (aaudiber) [inline common/src/test/java/tachyon/heartbeat/HeartbeatContextTest.java]: If you don't mind, can you add `HeartbeatContext.WORKER_BLOCK_SYNC` here a below as well?

**Comment by aaudiber** [comment]
Thanks for the improvement @shuhuai007! It's always great to see a bunch of duplicated code get removed :) I've left a few minor comments

File: common/src/main/java/tachyon/heartbeat/HeartbeatContext.java (hunk 1 above)
Comment (shuhuai007) [inline common/src/main/java/tachyon/heartbeat/HeartbeatContext.java]: Done

File: common/src/test/java/tachyon/heartbeat/HeartbeatContextTest.java (hunk 2 above)
Comment (shuhuai007) [inline common/src/test/java/tachyon/heartbeat/HeartbeatContextTest.java]: Done

**Comment by aaudiber** [comment]
Jenkins, test this please.

**Comment by aaudiber** [comment]
+1 LGTM

**Pull request merged**